### PR TITLE
Additional documentation related to nginx and `root_url`

### PR DESCRIPTION
**Pull request by abidlabs**
As I was looking at https://github.com/gradio-app/gradio/issues/7749, I realized we never really documented our process for constructing the `root_url` or explained what headers are needed in nginx. This add additional documentation.

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Looks good @abidlabs !

File: guides/09_other-tutorials/running-gradio-on-your-web-server-with-nginx.md [hunk 1]
```diff
@@ -42,15 +42,21 @@ server {
         proxy_set_header Upgrade $http_upgrade;
         proxy_set_header Connection "upgrade";
         proxy_set_header Host $host;
+        proxy_set_header X-Forwarded-Host $host;
+        proxy_set_header X-Forwarded-Proto $scheme;
     }
 }
 ```
 
+Tip: Setting the `X-Forwarded-Host` and `X-Forwarded-Proto` headers is important as Gradio uses these, in conjunction with the `root_path` parameter discussed below, to construct the public URL that your app is being served on. Gradio uses the public URL to fetch various static assets. If these headers are not set, your Gradio app may load in a broken state.
```
Comment (aliabd) [inline guides/09_other-tutorials/running-gradio-on-your-web-server-with-nginx.md:51]: Need to add an extra new line here.

Comment (aliabd) [reply guides/09_other-tutorials/running-gradio-on-your-web-server-with-nginx.md:51]:
```suggestion

Tip: Setting the `X-Forwarded-Host` and `X-Forwarded-Proto` headers is important as Gradio uses these, in conjunction with the `root_path` parameter discussed below, to construct the public URL that your app is being served on. Gradio uses the public URL to fetch various static assets. If these headers are not set, your Gradio app may load in a broken state.
```

**Review by aliabd — APPROVED** [review]
Looks good but please fix the new line before Tip

**Comment by abidlabs** [comment]
Thanks for the reviews guys!

**Pull request merged**